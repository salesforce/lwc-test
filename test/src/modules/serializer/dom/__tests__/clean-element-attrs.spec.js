/*
 * Copyright (c) 2018, salesforce.com, inc.
 * All rights reserved.
 * SPDX-License-Identifier: MIT
 * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
 */

describe('serialized element attributes', () => {
    it('should replace both id and for attributes with placeholders', () => {
        const elm = document.createElement('div');
        elm.innerHTML = `
            <label for="cheese">Do you like cheese?</label>
            <input type="checkbox" name="cheese" id="cheese">
        `.replace(/\s+/g, ' ');
        expect(elm).toMatchSnapshot();
    });

    it('should replace all id-referencing aria attributes with placeholders', () => {
        const elm = document.createElement('div');
        elm.innerHTML = `
            <div aria-activedescendant="aria-activedescendant">aria-activedescendant</div>
            <div aria-controls="aria-controls">aria-controls</div>
            <div aria-describedby="aria-describedby">aria-describedby</div>
            <div aria-details="aria-details">aria-details</div>
            <div aria-errormessage="aria-errormessage">aria-errormessage</div>
            <div aria-flowto="aria-flowto">aria-flowto</div>
            <div aria-labelledby="aria-labelledby">aria-labelledby</div>
            <div aria-owns="aria-owns">aria-owns</div>
        `.replace(/\s+/g, ' ');
        expect(elm).toMatchSnapshot();
    });

    it('should remove the lwc:host attribute', () => {
        const elm = document.createElement('div');
        elm.innerHTML = `
            <div lwc:host="12345">12345</div>
            <div lwc:host="abcde">abcde</div>
        `.replace(/\s+/g, ' ');
        expect(elm).toMatchSnapshot();
    });

    it('should replace the href attribute with a placeholder if it is a fragment id for anchor tags', () => {
        const elm = document.createElement('div');
        elm.innerHTML = `
            <a href="#ny-state-of-mind">ny state of mind</a>
            <a href="https://nystateofmind.com">ny state of mind</a>
        `.replace(/\s+/g, ' ');
        expect(elm).toMatchSnapshot();
    });

    it('should replace the href attribute with a placeholder if it is a fragment id for area tags', () => {
        const elm = document.createElement('div');
        elm.innerHTML = `
            <map name="kyoto">
                <area href="#teramachi"/>
                <area href="#kawaramachi"/>
                <area href="https://duckduckgo.com/?q=kyoto"/>
            </map>
        `.replace(/\s+/g, ' ');
        expect(elm).toMatchSnapshot();
    });

    it('should replace the href and xlink:href attributes with placeholders if it is a fragment id for svg use tags', () => {
        const elm = document.createElement('div');
        elm.innerHTML = `
            <svg width="100px" height="100px" viewport="0 0 100 100">
                <defs>
                    <circle id="black" r="10" cx="10" cy="10" fill="black"></circle>
                    <circle id="red" r="10" cx="14" cy="14" fill="red"></circle>
                </defs>
                <use href="#black"></use>
                <use xlink:href="#red"></use>
            </svg>
        `.replace(/\s+/g, ' ');
        expect(elm).toMatchSnapshot();
    });
});
