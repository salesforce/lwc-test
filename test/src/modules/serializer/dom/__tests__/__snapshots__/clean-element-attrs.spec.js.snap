// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`serialized element attributes should remove the lwc:host attribute 1`] = `
<div>
   
  <div>
    12345
  </div>
   
  <div>
    abcde
  </div>
   
</div>
`;

exports[`serialized element attributes should replace all id-referencing aria attributes with placeholders 1`] = `
<div>
   
  <div
    aria-activedescendant="[shadow:guid]"
  >
    aria-activedescendant
  </div>
   
  <div
    aria-controls="[shadow:guid]"
  >
    aria-controls
  </div>
   
  <div
    aria-describedby="[shadow:guid]"
  >
    aria-describedby
  </div>
   
  <div
    aria-details="[shadow:guid]"
  >
    aria-details
  </div>
   
  <div
    aria-errormessage="[shadow:guid]"
  >
    aria-errormessage
  </div>
   
  <div
    aria-flowto="[shadow:guid]"
  >
    aria-flowto
  </div>
   
  <div
    aria-labelledby="[shadow:guid]"
  >
    aria-labelledby
  </div>
   
  <div
    aria-owns="[shadow:guid]"
  >
    aria-owns
  </div>
   
</div>
`;

exports[`serialized element attributes should replace both id and for attributes with placeholders 1`] = `
<div>
   
  <label
    for="[shadow:guid]"
  >
    Do you like cheese?
  </label>
   
  <input
    id="[shadow:guid]"
    name="cheese"
    type="checkbox"
  />
   
</div>
`;

exports[`serialized element attributes should replace the href and xlink:href attributes with placeholders if it is a fragment id for svg use tags 1`] = `
<div>
   
  <svg
    height="100px"
    viewport="0 0 100 100"
    width="100px"
  >
     
    <defs>
       
      <circle
        cx="10"
        cy="10"
        fill="black"
        id="[shadow:guid]"
        r="10"
      />
       
      <circle
        cx="14"
        cy="14"
        fill="red"
        id="[shadow:guid]"
        r="10"
      />
       
    </defs>
     
    <use
      href="#[shadow:guid]"
    />
     
    <use
      xlink:href="#[shadow:guid]"
    />
     
  </svg>
   
</div>
`;

exports[`serialized element attributes should replace the href attribute with a placeholder if it is a fragment id for anchor tags 1`] = `
<div>
   
  <a
    href="#[shadow:guid]"
  >
    ny state of mind
  </a>
   
  <a
    href="https://nystateofmind.com"
  >
    ny state of mind
  </a>
   
</div>
`;

exports[`serialized element attributes should replace the href attribute with a placeholder if it is a fragment id for area tags 1`] = `
<div>
   
  <map
    name="kyoto"
  >
     
    <area
      href="#[shadow:guid]"
    />
     
    <area
      href="#[shadow:guid]"
    />
     
    <area
      href="https://duckduckgo.com/?q=kyoto"
    />
     
  </map>
   
</div>
`;
